shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/core/importer/DefaultImporter.java

```diff
@@ -144,12 +149,21 @@ private void doFlush(final DataSource dataSource, final List<DataRecord> buffer)
             connection.setAutoCommit(false);
             switch (buffer.get(0).getType()) {
                 case IngestDataChangeType.INSERT:
+                    if (null != rateLimitAlgorithm) {
+                        rateLimitAlgorithm.intercept(JobOperationType.INSERT, buffer.size());
+                    }
                     executeBatchInsert(connection, buffer);
                     break;
                 case IngestDataChangeType.UPDATE:
+                    if (null != rateLimitAlgorithm) {
+                        rateLimitAlgorithm.intercept(JobOperationType.UPDATE, buffer.size());
+                    }
                     executeUpdate(connection, buffer);
                     break;
                 case IngestDataChangeType.DELETE:
+                    if (null != rateLimitAlgorithm) {
+                        rateLimitAlgorithm.intercept(JobOperationType.DELETE, buffer.size());
+                    }
```

Comment:
It's better to use `1` instread of `buffer.size()`, since INSERT/UPDATE/DELETE is in transaction.